System/OsPath/Common.hs

```diff
@@ -1467,3 +1504,137 @@ decodeFS (OsString (PosixString x)) = decodeWithBasePosix x
 
 #endif
 
+#ifdef WINDOWS
+#elif defined(POSIX)
+#else
+
+-- | Get a list of 'FILEPATH's in the $PATH variable.
+getSearchPath :: IO [OsPath]
+#if defined(mingw32_HOST_OS) || defined(__MINGW32__)
```

Comment:
At line 1507 we checked whether WINDOWS is defined, presumably in our branch it isn't so is there any need to consider mingw32 platform here?

Reply:
Ok this function will only be visible in the `System.OsPath` module for the current platform. It will not be present in either `System.OsPath.Posix` nor `System.OsPath.Windows`.

Still it seems in such case it may be permissible for `filepath` to depend on the `Win32` package and reuse its environment handling functions.

Reply:
Win32 already depends on filepath, so we can't.

That's why I'm opting for https://github.com/haskell/directory/pull/198